pkgs/development/tools/misc/hydra/common.nix

```diff
@@ -36,107 +77,162 @@ let
-      perlDeps perl nix
+    };
+  in
+  stdenv.mkDerivation rec {
+    pname = "hydra";
+
+    inherit stdenv src version patches;
+
+    buildInputs = [
+      makeWrapper
+      autoconf
+      automake
+      libtool
+      unzip
+      nukeReferences
+      sqlite
+      libpqxx_6
+      top-git
+      mercurial
+      darcs
+      subversion
+      breezy
+      openssl
+      bzip2
+      libxslt
+      perlDeps
+      perl
+      nix
       postgresql # for running the tests
       nlohmann_json
       boost
+      pixz
+    ];
+
+    hydraPath = lib.makeBinPath ([
+      sqlite
+      subversion
+      openssh
+      nix
+      coreutils
+      findutils
+      pixz
+      gzip
+      bzip2
+      xz
+      gnutar
+      unzip
+      git
+      top-git
+      mercurial /*darcs*/
+      gnused
+      breezy
+    ] ++ lib.optionals stdenv.isLinux [ rpm dpkg cdrkit ]);
+
+    nativeBuildInputs = [ autoreconfHook pkg-config mdbook ];
+
+    configureFlags = [ "--with-docbook-xsl=${docbook_xsl}/xml/xsl/docbook" ];
+
+    checkInputs = [
+      foreman
+      python3
+      libressl.nc
     ];
 
-  hydraPath = lib.makeBinPath (
-    [ sqlite subversion openssh nix coreutils findutils pixz
-      gzip bzip2 xz gnutar unzip git top-git mercurial /*darcs*/ gnused breezy
-    ] ++ lib.optionals stdenv.isLinux [ rpm dpkg cdrkit ] );
-
-  nativeBuildInputs = [ autoreconfHook pkg-config mdbook ];
-
-  configureFlags = [ "--with-docbook-xsl=${docbook_xsl}/xml/xsl/docbook" ];
-
-  NIX_CFLAGS_COMPILE = "-pthread";
-
-  shellHook = ''
-    PATH=$(pwd)/src/script:$(pwd)/src/hydra-eval-jobs:$(pwd)/src/hydra-queue-runner:$(pwd)/src/hydra-evaluator:$PATH
-    PERL5LIB=$(pwd)/src/lib:$PERL5LIB;
-  '';
-
-  enableParallelBuilding = true;
-
-  preCheck = ''
-    patchShebangs .
-    export LOGNAME=''${LOGNAME:-foo}
-  '';
-
-  postInstall = ''
-    mkdir -p $out/nix-support
-    for i in $out/bin/*; do
-        read -n 4 chars < $i
-        if [[ $chars =~ ELF ]]; then continue; fi
-        wrapProgram $i \
-            --prefix PERL5LIB ':' $out/libexec/hydra/lib:$PERL5LIB \
-            --prefix PATH ':' $out/bin:$hydraPath \
-            --set HYDRA_RELEASE ${version} \
-            --set HYDRA_HOME $out/libexec/hydra \
-            --set NIX_RELEASE ${nix.name or "unknown"}
-    done
-  ''; # */
-
-  dontStrip = true;
-
-  passthru = { inherit perlDeps migration tests; };
-
-  meta = with lib; {
-    description = "Nix-based continuous build system";
-    license = licenses.gpl3;
-    platforms = platforms.linux;
-    maintainers = with maintainers; [ ma27 ];
-  };
-}
+    NIX_CFLAGS_COMPILE = "-pthread";
+
+    shellHook = ''
+      PATH=$(pwd)/src/script:$(pwd)/src/hydra-eval-jobs:$(pwd)/src/hydra-queue-runner:$(pwd)/src/hydra-evaluator:$PATH
+      PERL5LIB=$(pwd)/src/lib:$PERL5LIB;
+    '';
+
+    enableParallelBuilding = true;
+
+    postPatch = ''
+      # Remove some (seemingly?) broken tests.
+      rm \
+        t/queue-runner/notifications.t
```

Comment:
I think this is solved by https://github.com/NixOS/hydra/pull/1140 given this output:

```
[  FAIL  ]  job 54  +~Pre-build the job, upload to the cache, and then delete locally
[  FAIL  ]  job 54    + Reading the output's log succeeds
[  DEBUG ]  job 54    | queue-runner/notifications.t line 46
(  DIAG  )  job 54    | +-----+----+-------+
(  DIAG  )  job 54    | | GOT | OP | CHECK |
(  DIAG  )  job 54    | +-----+----+-------+
(  DIAG  )  job 54    | | 256 | eq | 0     |
(  DIAG  )  job 54    | +-----+----+-------+

```

I'm a bit shy to merge that PR, but I'll give it a shot early next week.